### ISSUE TITLE
Calling ResourceServers.GetAsync returns Not Found when identifier is a URL

### ISSUE THREAD

**Issue by WesleySSmith** [auth0/auth0.net#377]
### Description

I'm trying to fetch the definition of an API (ResourceServer) that has an Identifier of `https://example.com/example` But if I pass that identifier then I get a `Auth0.Core.Exceptions.ErrorApiException: Not Found` response.

### Reproduction

1. Create an API with a URL as the Identifier, such as "https://example.com/example".   (URLs are the suggested format for API Identifiers)

1. call `mClient.ResourceServers.GetAsync("https://example.com/example")`

Note that a Not Found exception is thrown.

A HTTP trace shows that the URL requested is 

`https://XXX.auth0.com/api/v2/resource-servers/https://example.com/example`

Running that same query via the Management API explorer shows that the correct URL should be:

`https://XXX.auth0.com/api/v2/resource-servers/https%3A%2F%2Fexample.com%2Fexample`

So, it seems that the id should be encoded before putting it in the URL.

Code that creates URL is in ResourceServersClient.cs:
```
 public Task<ResourceServer> GetAsync(string id)
        {
            return Connection.GetAsync<ResourceServer>(BuildUri($"resource-servers/{id}"), DefaultHeaders);
        }
```

### Environment

Version 7.0.2
dotnet core 3.1

**Comment by WesleySSmith** [comment]
Code inspection shows same problem would occur in DeleteAsync, and UpdateAsync as well.  And maybe other clients as well?

**Comment by damieng** [comment]
yeah, applying the fix broadly.

**Comment by WesleySSmith** [comment]
@damieng In the commit referenced, it looks like Uri.EscapeUriString was used, but in my experience, that doesn't do what you want.  Uri.EscapeDataString is the right one to use.

**Comment by damieng** [comment]
You're totally right and I was missing the test coverage that would indicate that.  Addressed in #379
